src/backend/allocator/vulkan/mod.rs

```diff
@@ -184,13 +185,20 @@ impl VulkanAllocator {
     ///
     /// # Panics
     ///
-    /// If the version of instance which created the [`PhysicalDevice`] is higher than [`VulkanAllocator::MAX_INSTANCE_VERSION`].
+    /// - If the version of instance which created the [`PhysicalDevice`] is higher than [`VulkanAllocator::MAX_INSTANCE_VERSION`].
+    /// - If the default [`ImageUsageFlags`] are empty.
     pub fn new(phd: &PhysicalDevice, default_usage: ImageUsageFlags) -> Result<VulkanAllocator, Error> {
         // Panic if the instance version is too high
         if phd.instance().api_version() > Self::MAX_INSTANCE_VERSION {
             panic!("Exceeded maximum instance api version for VulkanAllocator (1.3 max)")
         }
 
+        // VUID-VkPhysicalDeviceImageFormatInfo2-usage-requiredbitmask
+        // At least one image usage flag must be specified.
+        if default_usage.is_empty() {
+            panic!("Default usage flags for allocator are empty")
```

Comment:
I mean this error is likely fatal somewhere higher up, but couldn't we just return an error in this case? What is the rationale behind panic'ing here?

Reply:
My rationale was that generally you'd know ahead of time what default usages you would need to support.

Reply:
Yeah that seems reasonable